packages/perseus/src/widgets/grapher/grapher-validator.ts

```diff
@@ -20,21 +20,36 @@ function grapherValidator(
     }
 
     // We haven't moved the coords
-    if (state.coords == null) {
+    if (userInput.coords == null) {
         return {
             type: "invalid",
             message: null,
         };
     }
 
     // Get new function handler for grading
-    const grader = functionForType(state.type);
-    const guessCoeffs = grader.getCoefficients(state.coords, state.asymptote);
-    const correctCoeffs = grader.getCoefficients(
-        rubric.correct.coords,
-        // @ts-expect-error - TS(2339) - Property 'asymptote' does not exist on type '{ type: "absolute_value"; coords: [Coord, Coord]; }'
-        rubric.correct.asymptote,
-    );
+    const grader = functionForType(userInput.type);
```

Comment:
I was hoping we wouldn't be changing the validation logic so much as describing the validation logic with proper types. This looks correct, but how confident are we? Is there test coverage for this?